packages/core/types/Config.tsx

```diff
@@ -63,6 +63,13 @@ export type ComponentConfig<
         props?: Partial<ComponentProps>;
         readOnly?: Partial<Record<keyof ComponentProps, boolean>>;
       };
+  resolvePermissions?: (
+    data: DataShape,
+    params: {
+      lastPermissions: Partial<Permissions> | undefined;
+      initialPermissions: Partial<Permissions>;
```

Comment:
I think we're missing a couple of things off the params that would keep this in sync with the other resolvers

* `changed` - An object describing which props have changed on this component since the last time this function was called. The [`get-changed`](https://github.com/measuredco/puck/blob/main/packages/core/lib/get-changed.ts) lib has your back here.
* `appState` - current app state